### PR TITLE
user can add connection requests

### PR DESCRIPTION
**Pull request by lspdrz**
## Proposed changes

Adds a POST endpoint at /users/:id/connections for users to add connection requests.

Also removes facilityId from Connection, as well as adds an @Unique decorator based on the inmateId and connectionId that ensures there is only one Connection for any combination of inmateId and connectionId.

## Related Issues
i.e Closes Issue #16 

## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

### CODE REVIEW

File: src/connections/entities/connection.entity.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 import Inmate from '@src/inmates/entities/inmate.entity';
 import User from '@src/users/entities/user.entity';
-import { Entity, Column, ManyToOne } from 'typeorm';
+import { Entity, Column, ManyToOne, Unique } from 'typeorm';
 import { ConnectionStatus } from '@src/connections/enums/connectionStatus.enum';
 import ConnectionRelationship from '../enums/connectionRelationship.enum';
-import Facility from '@src/facilities/entities/facility.entity';
 
 @Entity({ name: 'Connection' })
+@Unique('unique_connection', ['inmateId', 'userId'])
```
Comment (gabrielsaruhashi) [inline src/connections/entities/connection.entity.ts:9]: this is really cool

File: src/facilities/facilities.controller.ts [hunk 2]
```diff
@@ -67,7 +67,7 @@ export default class FacilitiesController {
   }
 
   @Get(':id/connections')
-  @UseGuards(StaffGuard)
+  // @UseGuards(StaffGuard)
```
Comment (gabrielsaruhashi) [inline src/facilities/facilities.controller.ts:70]: should we uncomment this?

File: src/facilities/facilities.service.ts [hunk 3]
```diff
@@ -56,9 +63,19 @@ export default class FacilitiesService {
...
 
   async getFacilityConnections(id: number) {
-    const facility = await this.getFacilityById(id);
+    const facility = await getRepository(Facility)
+      .createQueryBuilder('facility')
+      .innerJoinAndSelect('facility.inmates', 'inmates')
+      .innerJoinAndSelect('inmates.connections', 'connections')
+      .where(`facility.id = ${id}`)
+      .getOne();
+    let connections: Connection[] = [];
```
Comment (gabrielsaruhashi) [inline src/facilities/facilities.service.ts:72]: lets just make this one line
const connections =  facility.inmates.map((inmate) => inmate.connections);

File: src/facilities/facilities.service.ts [hunk 4]
```diff
@@ -56,9 +63,19 @@ export default class FacilitiesService {
...
+    facility.inmates.map((inmate) => {
+      connections = connections.concat(inmate.connections);
+    });
+    return connections;
     // TODO: invoke buildUserRo when returning the user
-    return facility.connections.map((c) => ({ ...c, user: c.user }));
+    // return facility.connections.map((c) => ({ ...c, user: c.user }));
```
Comment (gabrielsaruhashi) [inline src/facilities/facilities.service.ts:78]: if we remove the user, can we create a GetFacilityConnectionUsers or smt to get the info of all the connections' user info?

File: src/users/users.controller.ts [hunk 5]
```diff
@@ -27,6 +34,19 @@ export default class UsersController {
     return { connections: await this.usersService.getUserConnections(id) };
   }
 
+  @Post(':id/connections')
+  @UseGuards(UserGuard)
+  async createUserConnection(
+    @Param('id', ParseIntPipe) id: number,
```
Comment (gabrielsaruhashi) [inline src/users/users.controller.ts:40]: no need to ParseIntPipe here
we're using aa globalPipeTransform that does all these checks based on the type

File: src/users/users.service.ts [hunk 6]
```diff
@@ -53,6 +54,10 @@ export default class UsersService {
     return user.connections;
   }
 
+  // async createConnection(connection: CreateConnectionDto) {
```
Comment (gabrielsaruhashi) [inline src/users/users.service.ts:57]: delete?

**Review by gabrielsaruhashi — CHANGES_REQUESTED** [review]
some nit requests! but looks great 🔥

**Review by gabrielsaruhashi — APPROVED** [review]
actually nvm lol
I'll just stamp! Im just asking for some nit changes

File: src/facilities/facilities.service.ts (hunk 3 above)
Comment (lspdrz) [reply src/facilities/facilities.service.ts:72]: That would return a double array of connections, where I want to concat elements from each array to the main connections array. I'm adding a TODO for now so we can revisit

**Pull request merged**